File: designPattern/Builder/python/Builder.py
import abc


class Builder:
    __metaclass__ = abc.ABCMeta

    def setTitle(self, title):
        pass

    def setText(self, text):
        pass

    def setItems(self, items):
        pass

    def close(self):
        pass

    def getResult(self):
        pass

File: designPattern/Builder/reviewJava/Builder.java
public abstract class Builder{
	StringBuilder strBuilder = new StringBuilder();
	public abstract void makeTitle(String title);
	public abstract void makeItem(String[] items);
	public abstract void makeText(String text);
	public abstract void close();
	public void appendStr(String text){
		strBuilder.append(text).append("\n\n");
	}
	public String result(){
		return strBuilder.toString();
	}
}